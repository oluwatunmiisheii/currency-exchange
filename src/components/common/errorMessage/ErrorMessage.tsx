import React from 'react'
import { InformationCircleIcon } from '@heroicons/react/solid';


interface IErrorMessage {
  message?: string;
}
 
const ErrorMessage: React.FC<IErrorMessage> = ({ message }) => {
  return ( 
    <small className="text-sm flex items-center mt-2">
      <InformationCircleIcon className="h-4 w-4 text-red-500 mr-1" /> 
      <span className="text-gray-600">{message}</span>
    </small>
   );
}
 
export default ErrorMessage;