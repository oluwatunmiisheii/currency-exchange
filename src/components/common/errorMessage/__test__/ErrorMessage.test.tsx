import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorMessage from '../ErrorMessage';
 
const errorMessage = 'This is an error message';
const setup = (props: {message: string}) => render(<ErrorMessage {...props} />);

it('should render error message passed as props', async () => {
  const { container } = setup({ message: errorMessage });

  const message = screen.getByText(/^This is an error message$/i); // full string match, ignore case

  expect(message).toBeInTheDocument();
  expect(message).toHaveTextContent(errorMessage);

  expect(container).toMatchSnapshot();
});