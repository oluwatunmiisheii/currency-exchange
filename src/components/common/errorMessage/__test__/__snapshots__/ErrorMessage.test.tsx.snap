// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render error message passed as props 1`] = `
<div>
  <small
    class="text-sm flex items-center mt-2"
  >
    <svg
      aria-hidden="true"
      class="h-4 w-4 text-red-500 mr-1"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class="text-gray-600"
    >
      This is an error message
    </span>
  </small>
</div>
`;
